webapp/components/quick_switch_modal/quick_switch_modal.jsx

```diff
@@ -0,0 +1,323 @@
+
+import $ from 'jquery';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {browserHistory} from 'react-router/es6';
+import {Modal} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+// Redux actions
+import store from 'stores/redux_store.jsx';
+const getState = store.getState;
+
+import {getChannel} from 'mattermost-redux/selectors/entities/channels';
+import {getUserByUsername} from 'mattermost-redux/selectors/entities/users';
+
+const CHANNEL_MODE = 'channel';
+const TEAM_MODE = 'team';
+
+export default class QuickSwitchModal extends React.PureComponent {
+    static propTypes = {
+
+        /**
+         * The mode to start in when showing the modal, either 'channel' or 'team'
+         */
+        initialMode: PropTypes.string.isRequired,
+
+        /**
+         * Set to show the modal
+         */
+        show: PropTypes.bool.isRequired,
+
+        /**
+         * The function called to hide the modal
+         */
+        onHide: PropTypes.func.isRequired,
+
+        /**
+         * Set to show team switcher
+         */
+        showTeamSwitcher: PropTypes.bool
+    }
+
+    static defaultProps = {
+        initialMode: CHANNEL_MODE
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.onChange = this.onChange.bind(this);
+        this.onShow = this.onShow.bind(this);
+        this.onHide = this.onHide.bind(this);
+        this.onExited = this.onExited.bind(this);
+        this.handleKeyDown = this.handleKeyDown.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.switchToChannel = this.switchToChannel.bind(this);
+        this.switchMode = this.switchMode.bind(this);
+        this.focusTextbox = this.focusTextbox.bind(this);
+
+        this.enableChannelProvider = this.enableChannelProvider.bind(this);
+        this.enableTeamProvider = this.enableTeamProvider.bind(this);
+        this.channelProviders = [new SwitchChannelProvider()];
+        this.teamProviders = [new SwitchTeamProvider()];
+
+        this.state = {
+            text: '',
+            mode: props.initialMode
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.show && !prevProps.show) {
+            this.focusTextbox();
+        }
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (!this.props.show && nextProps.show) {
+            this.setState({mode: nextProps.initialMode, text: ''});
+        }
+    }
+
+    focusTextbox() {
+        if (this.refs.switchbox == null) {
+            return;
+        }
+
+        const textbox = this.refs.switchbox.getTextbox();
+        textbox.focus();
+        Utils.placeCaretAtEnd(textbox);
+    }
+
+    onShow() {
+        this.setState({
+            text: ''
+        });
+    }
+
+    onHide() {
+        this.setState({
+            text: ''
+        });
+        this.props.onHide();
+    }
+
+    onExited() {
+        setTimeout(() => {
+            $('#post_textbox').get(0).focus();
```

Comment:
This was existing code I just moved around but I can try to find a workaround to not use jquery